Source/Controllers/Other/SecureBookmarkData.swift

```diff
@@ -8,61 +8,58 @@
 
 import Foundation
 
-class SecureBookmarkData: NSObject, NSCoding, NSSecureCoding {
- 
-    var bookmarkData: Data
-    var options: Double
-    var theUrl: URL
-    
-    static var supportsSecureCoding: Bool {
-        return true
-    }
-   
-    init(data: Data, options: Double, url: URL ) {
-        self.bookmarkData = data
+final class SecureBookmarkData: NSObject {
+    internal let bookmarkData: Data
```

Comment:
internal is also used only for framework, not necessary:

```
- Open access and public access enable entities to be used within any source file from their defining module, and also in a source file from another module that imports the defining module. You typically use open or public access when specifying the public interface to a framework. The difference between open and public access is described below.
- Internal access enables entities to be used within any source file from their defining module, but not in any source file outside of that module. You typically use internal access when defining an app’s or a framework’s internal structure.
- File-private access restricts the use of an entity to its own defining source file. Use file-private access to hide the implementation details of a specific piece of functionality when those details are used within an entire file.
- Private access restricts the use of an entity to the enclosing declaration, and to extensions of that declaration that are in the same file. Use private access to hide the implementation details of a specific piece of functionality when those details are used only within a single declaration.
```

TLDR - in project, you only use `private` or `private(set)`, nothing else :)

Reply:
bookmarkData is read in the manager ..... so what should that be?

Reply:
Just need help with this one please @Kaspik

Reply:
nothing, simply `let` which makes it available for other classes